kernel/rule.mli

```diff
@@ -49,16 +49,19 @@ val pp_rule_name : rule_name printer
 type 'a rule =
   {
     name: rule_name;
-    ctx : 'a;
+    ctx : 'a context;
     pat : pattern;
     rhs : term
   }
 
 val get_loc_rule : 'a rule -> loc
 
-type untyped_rule = untyped_context rule
-
-type typed_rule = typed_context rule
+type part_typed_rule = term option rule
```

Comment:
partially_typed_rule instead? But the name of the type is still not satisfactory because we are refering to the context which is not really explicit.